src/adt/edgeset.c

```diff
@@ -349,12 +349,15 @@ edge_set_remove(struct edge_set **setp, unsigned char symbol)
 		return;
 	}
 
-	for (i = 0; i < set->i; i++) {
+	i = 0;
+	while (i < set->i) {
 		if (set->a[i].symbol == symbol) {
 			if (i < set->i) {
 				memmove(&set->a[i], &set->a[i + 1], (set->i - i - 1) * (sizeof *set->a));
 			}
 			set->i--;
+		} else {
+			i++;
```

Comment:
Amazing that despite all the tests, we still didn't hit this

Reply:
For what it's worth, the way I found this was using `gengraph` to generate several corpora of NFAs in ascending sizes, used `fsm -pd` to turn them into DFAs, and then running the new (work in progress) minimization code on them. Some `assert` failures started showing up, and with a corpus of several thousand more small DFAs I eventually got a few that  triggered the asserts, yet were small enough (<10 states or so) to understand and reduce by hand.